### PR TITLE
AP-2816 Send employment data to CFE

### PR DESCRIPTION
**Pull request by stephenrichards**
## Send employment data to CFE

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2816)

The non-passported jouney now needs to send employment data to CFE.

- Added `CFE::CreateEmploymentsService` to send employment data to CFE
- Moved the classes which parse the HMEC response from `app/models` to `services/hmrc/parsed_response` which makes more sense
- Added new state `employments_created` to `CFESubmissionStateMachine`
- Ensured `CreateCapitalsService` only sent values of 2 decimal places to CFE (feature tests were failing because it was randomly sending numbers to 8 places of decimals)
- Added rake task `employ` which only works in development and will set up the database for the employed journey.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/concerns/cfe_submission_state_machine.rb [hunk 1]
```diff
@@ -62,8 +63,12 @@ module CFESubmissionStateMachine
         transitions from: :other_income_created, to: :irregular_income_created, guard: :non_passported?
```
Comment (colinbruce) [inline app/models/concerns/cfe_submission_state_machine.rb:63]: Should this transition to `employments_created`?

File: spec/factory_helpers/hmrc_response/use_case_one.rb [hunk 2]
```diff
@@ -42,13 +43,147 @@ def response
...
         }
       end
 
       private
 
-      def data_array
+      def generate_named_data
+        case @options[:named_data]
+        when :eg1_uc1
+          eg1_uc1
```
Comment (colinbruce) [inline spec/factory_helpers/hmrc_response/use_case_one.rb:55]: Naming things is hard... does eg1 mean example1?  
I wasn't sure if there was another, hidden, meaning

File: spec/services/cfe/create_cash_transactions_service_spec.rb [hunk 3]
```diff
@@ -58,8 +58,8 @@ module CFE
       context 'successful post' do
         before { stub_request(:post, service.cfe_url).with(body: expected_payload_hash.to_json).to_return(body: dummy_response) }
 
-        it 'updates the submission record from irregular_income_created to cash_transactions_created' do
-          expect(submission.aasm_state).to eq 'irregular_income_created'
+        it 'updates the submission record from employments_created to cash_transactions_created' do
```
Comment (colinbruce) [inline spec/services/cfe/create_cash_transactions_service_spec.rb:61]: May need an extra test here for the transition _to_ employments_created

**Review by colinbruce — CHANGES_REQUESTED** [review]
Looking good :+1:
A couple of questions and one thing I _think_ needs changing

File: app/models/concerns/cfe_submission_state_machine.rb (hunk 1 above)
Comment (stephenrichards) [reply app/models/concerns/cfe_submission_state_machine.rb:63]: No, I don't think so.  It's the `irregular_income_created` event, so it transitions to `irregular_income_created`.  The `employments_created` event transitions to `employments_created`.

File: spec/factory_helpers/hmrc_response/use_case_one.rb (hunk 2 above)
Comment (stephenrichards) [reply spec/factory_helpers/hmrc_response/use_case_one.rb:55]: Yes, - I'll rename it to `example1_usecase1`

File: app/models/concerns/cfe_submission_state_machine.rb (hunk 1 above)
Comment (colinbruce) [reply app/models/concerns/cfe_submission_state_machine.rb:63]: Yes, I see it now! :+1:

File: spec/services/cfe/create_cash_transactions_service_spec.rb (hunk 3 above)
Comment (colinbruce) [reply spec/services/cfe/create_cash_transactions_service_spec.rb:61]: Ignorable :+1:

**Pull request merged**